### PR TITLE
model comparison gollm cache issue

### PR DESCRIPTION
**Pull request by YohannParis**
# Description

Resolves #(issue)

### CODE REVIEW

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue [hunk 1]
```diff
@@ -290,30 +289,28 @@ async function buildJupyterContext() {
...
-	// Generate once the comparison task has been completed
-	if (!compareModelsTaskOutput) return;
-	assignOverview(compareModelsTaskOutput);
-	emit('update-status', OperatorStatus.DEFAULT); // This is a custom way of granting a default status to the operator, since it has no output
-}
-
+// Listen for the task completion event
 useClientEvent(ClientEventType.TaskGollmCompareModel, (event: ClientEvent<TaskResponse>) => {
-	if (!event.data || event.data.id !== compareModelsTaskId) return;
+	if (!isEmpty(overview.value) && event?.data.id === compareModelsTaskId) return;
```
Comment (jryu01) [inline packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: Just curious, are there cases where an event with a takeId is fired after an overview has already been generated from the task output with the same taskId?

Comment (YohannParis) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: Nope, there is not, but there is a situation were the request to get an overview is answered before the client-event does and therefore there is no need to overwrite it twice.

[the remark this answers is not in the document]
Comment (dgauldie) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: better yet, you can or them together into a single if statement
```
if (!event.data 
	|| event.data.id !== compareModelsTaskId 
	|| !isEmpty(overview.value) 
	|| event.data.status !== TaskStatus.Success
) {
	return;
}
```

[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: Those are not the same test. One check that the overview has already been generated for this specific taskId.
The others are for a normal check.

[the remark this answers is not in the document]
Comment (dgauldie) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: it is the same test

[the remark this answers is not in the document]
Comment (dgauldie) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue:309]: why have 4 separate if statements when you can roll into 1

**Review by dgauldie — APPROVED** [review]
LGTM

File: packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (YohannParis) [reply packages/client/hmi-client/src/components/workflow/ops/model-comparison/tera-model-comparison.vue]: Updated

**Pull request merged**